chain/ethereum/subscription.go

```diff
@@ -217,11 +217,13 @@ func (s *Subscription) subscribeHeaders(ctx context.Context) {
 		case header := <-headers:
 			//	Start a backfill when there are missing blocks
 			if s.latestBlockNumber != nil && header.Number.Uint64()-s.latestBlockNumber.Uint64() > 1 {
-				if backfillLogs, err := HistoricalEvents(ctx, s.client, s.addresses, s.latestBlockNumber.Uint64(), header.Number.Uint64()); err == nil {
-					for bfLog := range backfillLogs {
-						s.inLogs <- bfLog
+				go func(c context.Context, cl *ethclient.Client, adr []common.Address, fromBl, toBl uint64) {
```

Comment:
Approved but this function can be re-written to improve readability. Only latestBlockNumber will be changed so you  don't need to pass other variables to the child goroutine by value as well.

There is another way to pass an object to an anonymous function by value you might not know. It can be written like this:
```
var v int // A variable that might be changed by other goroutine

func someFunction() {
	copy := v // A copy of v
	
	go func ()  {
		// Now value of v can be safely used inside the goroutine without worrying about v being modified
		doSomethingWithValueOfV(copy)
	}()
}
```